app-alternatives/pinentry/pinentry-0.ebuild

```diff
@@ -0,0 +1,33 @@
+# Copyright 2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+ALTERNATIVES=(
+	"ncurses:app-crypt/pinentry[ncurses]"
+	"gtk:app-crypt/pinentry[gtk]"
+	"qt5:app-crypt/pinentry[qt5]"
+	"efl:app-crypt/pinentry[efl]"
+	"tty:app-crypt/pinentry"
```

Comment:
This will cause conflicting flags in the default configuration for users of any desktop profile:
- `ncurses` is enabled in `profiles/default/linux`
- both `gtk` and `qt5` are enabled in `profiles/targets/desktop`

Also, `eselect-pinentry` supports 6 different symlink targets. Why are only 5 of them listed here?

Reply:
> This will cause conflicting flags in the default configuration for users of any desktop profile:
> 
>     * `ncurses` is enabled in `profiles/default/linux`
> 
>     * both `gtk` and `qt5` are enabled in `profiles/targets/desktop`

That's why I disabled them in `profiles/targets/desktop/package.use`. If this doesn't work, how should I manage that?

> Also, `eselect-pinentry` supports 6 different symlink targets. Why are only 5 of them listed here?

This was probably a mistake, will fix.

Reply:
IMHO it is confusing behaviour if a package doesn't accept the default USE flags that are enabled in the profile, and the user needs to work around it (and reconfirm a flag that for all other packages is enabled by default). AFAICS no other `app-alternatives` package does this. Also, what if the user has set a flag like `gtk` or `qt5` in `make.conf`?

In summary, I think that especially for beginners it won't be obvious how to configure this correctly, namely explicitly enable one of the flags in `package.use` and make sure that all others are disabled. Compare that with `eselect pinentry` whose usage is self-explanatory.

Reply:
Since all of the alternatives are within a single package why not control the default symlink there without requiring app-alternatives?

Reply:
> Since all of the alternatives are within a single package why not control the default symlink there without requiring app-alternatives?

That's probably the worst alternative, because it would require recompilation of the package (i.e. `app-crypt/pinentry`) just to change the symlink. In comparison, reinstalling the alternatives package is cheap.

Reply:
> IMHO it is confusing behaviour if a package doesn't accept the default USE flags that are enabled in the profile, and the user needs to work around it (and reconfirm a flag that for all other packages is enabled by default). AFAICS no other `app-alternatives` package does this. Also, what if the user has set a flag like `gtk` or `qt5` in `make.conf`?
> 
> In summary, I think that especially for beginners it won't be obvious how to configure this correctly, namely explicitly enable one of the flags in `package.use` and make sure that all others are disabled. Compare that with `eselect pinentry` whose usage is self-explanatory.

Yeah, I understand what you're getting at. I agree this may not be the best approach. I won't close this yet, but I agree with you here.